### PR TITLE
Add [dependabot] SemVer compatibility badge

### PR DESCRIPTION
**Pull request by greysteil**
We currently make a request to the generic badge endpoint to generate these - would be rad to serve them directly from shields.io.

You can see them in action [here](https://dependabot.com/compatibility-score.html?dependency-name=puma&package-manager=bundler&version-scheme=semver). Use case is library maintainers putting them in their readmes.

### CODE REVIEW

**Comment by greysteil** [comment]
@platan - I don't think the failing test here is anything to do with my badge, would you mind asking CI to re-run? (I don't have permission.)

**Comment by platan** [comment]
Not a problem :smile: [Tests are OK](https://circleci.com/gh/badges/shields/5833?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

**Comment by greysteil** [comment]
Thanks! ❤️

**Comment by greysteil** [comment]
@paulmelnikow - long awaited follow-up to https://github.com/badges/shields/pull/1516 (which I'll close - I've included the logo here). 🎉

File: server.js [hunk 1]
```diff
@@ -7965,6 +7965,40 @@ cache(function (data, match, sendBadge, request) {
   });
 }));
 
+// Dependabot SemVer compatibility integration
+camp.route(/^\/dependabot\/semver\/([^/]+)\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var package_manager = match[1];
```
Comment (platan) [inline server.js]: Can we use `camelCase` notation here (and below) instead of `snake_case`?

Comment (platan) [inline server.js]: We can use `const` instead of `var` in all paces.

File: server.js [hunk 2]
```diff
@@ -7965,6 +7965,40 @@ cache(function (data, match, sendBadge, request) {
...
+  var options = {
+    method: 'GET',
+    headers: { 'Accept': 'application/json' },
+    uri: `https://api.dependabot.com/badges/compatibility_score?package-manager=${package_manager}&dependency-name=${dependency_name}&version-scheme=semver`
+  };
+  var badgeData = getBadgeData('semver compatibility', data);
+  badgeData.links = [`https://dependabot.com/compatibility-score.html?package-manager=${package_manager}&dependency-name=${dependency_name}&version-scheme=semver`];
+  badgeData.logo = getLogo('dependabot', data);
+  request(options, function(err, res) {
+    if (err != null) {
```
Comment (platan) [inline server.js]: `checkErrorResponse` is a great way to handle errors. It's described [here](https://github.com/badges/shields/blob/master/doc/TUTORIAL.md).

File: server.js [hunk 3]
```diff
@@ -7965,6 +7965,40 @@ cache(function (data, match, sendBadge, request) {
...
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var dependabotData = JSON.parse(res['body']);
+      badgeData.text[1] = dependabotData.status;
+      badgeData.colorscheme = dependabotData.colour;
+      sendBadge(format, badgeData);
+    } catch(e) {
+      badgeData.text[1] = 'error';
```
Comment (platan) [inline server.js]: In most of the cases we have `invalid` here (compare with https://github.com/badges/shields/blob/master/doc/TUTORIAL.md)

File: services/dependabot/dependabot.tester.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+const { isIntegerPercentage } = require('../test-validators');
+const colorscheme = require('../../lib/colorscheme.json');
+const mapValues = require('lodash.mapvalues');
+const colorsB = mapValues(colorscheme, 'colorB');
+
+const t = new ServiceTester({ id: 'dependabot', title: 'Dependabot' });
+module.exports = t;
+
+t.create('semver compatibility (valid)')
+  .get('/semver/bundler/puma.json')
```
Comment (platan) [inline services/dependabot/dependabot.tester.js]: You have to add `?style=_shields_test` at the end of url (`/semver/bundler/puma.json?style=_shields_test`) if you want to test colors. Without it JSON response does not contain `colorB` property. `Joi.equal(colorsB.brightgreen, colorsB.orange)` is true when property is missing. `Joi.equal(colorsB.brightgreen, colorsB.orange).required()` means that property is required.

File: services/dependabot/dependabot.tester.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  .expectJSONTypes(Joi.object().keys({
+    name: 'semver compatibility',
+    value: isIntegerPercentage,
+    link: 'https://dependabot.com/compatibility-score.html?dependency-name=puma&package-manager=bundler&version-scheme=semver',
+    colorB: Joi.equal(colorsB.brightgreen, colorsB.orange)
+  }));
+
+t.create('semver compatibility (connection error)')
```
Comment (platan) [inline services/dependabot/dependabot.tester.js]: Could you please add more test cases (missing dependency, invalid response). https://github.com/badges/shields/blob/master/doc/service-tests.md contains some examples.

**Comment by greysteil** [comment]
Thanks for the review @platan. Updated!

**Comment by greysteil** [comment]
Thanks for the review @chris48s. What's the process to get this merged from here? Anything I can do? :octocat:

**Comment by chris48s** [comment]
I'll leave it open to see if @platan has any final comments as I don't think he's checked back since the initial comment. If there's nothing else one of us will merge it.

File: server.js [hunk 6]
```diff
@@ -7965,6 +7965,38 @@ cache(function (data, match, sendBadge, request) {
...
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      const dependabotData = JSON.parse(res['body']);
+      badgeData.text[1] = dependabotData.status;
+      badgeData.colorscheme = dependabotData.colour;
+      sendBadge(format, badgeData);
+    } catch(e) {
+      badgeData.text[1] = 'invalid';
```
Comment (platan) [inline server.js]: @greysteil This fragment of code does not have tests. Would you like to add one? You can use mocked response with invalid JSON. An example: https://github.com/badges/shields/blob/ea4b758612c6e87ba227773fddee7ceab03ee138/services/bountysource/bountysource.tester.js#L30-L36

**Comment by greysteil** [comment]
@platan - done! Good to go? 🙂

**Comment by greysteil** [comment]
🎉 thanks!

Looks like it's been a few months since a deploy - anything I can do to help unblock that? :octocat:

**Pull request merged**